src/main/java/parsing/Parser.java

```diff
@@ -55,36 +61,74 @@ public NodeGraph parse(final String filename, NodeGraph graph) {
             BufferedReader in = new BufferedReader(
                     new InputStreamReader(getClass().getResourceAsStream(filename)));
             String line = in.readLine();
+            boolean newCache = true;
+            line = line.substring(line.indexOf("\t") + 1);
+            line = line.replaceAll(":", "");
+            String workingDirectory = System.getProperty("user.dir");
 
-            while (line != null) {
+            String absoluteFilePath;
+
+            absoluteFilePath = workingDirectory + File.separator + line;
+            String cacheName = line;
+            File file = new File(absoluteFilePath + ".txt");
```

Comment:
Using the header as file name will result in issues since in some files, the header is too large to be used as file name.